runtime-modules/referendum/src/lib.rs

```diff
@@ -685,6 +688,23 @@ impl<T: Trait<I>, I: Instance> EnsureChecks<T, I> {
     }
 
     fn can_vote(origin: T::Origin, stake: &Balance<T, I>) -> Result<T::AccountId, Error<T, I>> {
+        fn get_existing_stake<T: Trait<I>, I: Instance>(
```

Comment:
I don't understand what is going on here.

Why are we returning the stake of a vote that was for a old cycle? Why do we care about that?
Isn't `can_vote` just supposed to say `yes` if the vote does not exist, or is from an old cycle, and `no` otherwise? This is what `Mutations::vote` does.
If it returned `yes`, then the voting should just overwrite the vote and the lock, no?

Reply:
I mistakenly thought about `total_balance()` as if it was `free_balance()`; subtracting existing locked stake value from this ~~free_balance~~. It has been fixed - no need to retrieve existing stake now.